src/gluonts/nursery/duru/data.py

```diff
@@ -0,0 +1,488 @@
+import numpy as np
+import sklearn.model_selection
+import pickle
+import os
+import torchvision.transforms as transforms
+from gluonts.dataset.repository.datasets import get_dataset, dataset_recipes
+from gluonts.dataset.split import OffsetSplitter
+from utils import calculate_padding_base_2
+from gluonts.dataset.rolling_dataset import StepStrategy, NumSplitsStrategy
+from gluonts.dataset.rolling_dataset import generate_rolling_dataset
+
+
+def set_up_data(H):
+    # TODO split into train/val/test
+
+    if H.dataset == "dummy":
+        n_samples_train_val = 1000
+        n_samples_val = 200
+        n_samples_test = 500
+        # -
+        H.n_meas = 5
+        n_steps = 10000
+
+        # Version 1): random
+        X_train = (
+            torch.empty(n_samples_train_val, H.n_meas, n_steps)
+            .normal_()
+            .float()
+            .numpy()
+        )  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        Y_train = torch.empty(n_samples_train_val).bernoulli_().float().numpy()
+        X_test = (
+            torch.empty(n_samples_test, H.n_meas, n_steps)
+            .normal_()
+            .float()
+            .numpy()
+        )  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        Y_test = torch.empty(n_samples_test).bernoulli_().float().numpy()
+        # Version 2): all ones
+        # X_train = torch.ones(n_samples_train_val, H.n_meas, n_steps).float().numpy()  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        # Y_train = torch.ones(n_samples_train_val).float().numpy()
+        # X_test = torch.ones(n_samples_test, H.n_meas, n_steps).float().numpy()  # Conv1D expects (N,C_in,L_in), where C_in=number of channels=measurement dimension, L_in=length of signal sequence
+        # Y_test = torch.ones(n_samples_test).float().numpy()
+
+    elif H.dataset in list(
+        dataset_recipes.keys()
+    ):  # all available datasets in gluonts: ['constant', 'exchange_rate', 'solar-energy', 'electricity', 'traffic', 'exchange_rate_nips', 'electricity_nips', 'traffic_nips', 'solar_nips', 'wiki-rolling_nips', 'taxi_30min', 'kaggle_web_traffic_with_missing', 'kaggle_web_traffic_without_missing', 'kaggle_web_traffic_weekly', 'm1_yearly', 'm1_quarterly', 'm1_monthly', 'nn5_daily_with_missing', 'nn5_daily_without_missing', 'nn5_weekly', 'tourism_monthly', 'tourism_quarterly', 'tourism_yearly', 'cif_2016', 'london_smart_meters_without_missing', 'wind_farms_without_missing', 'car_parts_without_missing', 'dominick', 'fred_md', 'pedestrian_counts', 'hospital', 'covid_deaths', 'kdd_cup_2018_without_missing', 'weather', 'm3_monthly', 'm3_quarterly', 'm3_yearly', 'm3_other', 'm4_hourly', 'm4_daily', 'm4_weekly', 'm4_monthly', 'm4_quarterly', 'm4_yearly', 'm5', 'uber_tlc_daily', 'uber_tlc_hourly', 'airpassengers']
```

Comment:
I think we can remove that comment. Available datasets might change in the feature.

Reply:
If I was interested in knowing what datasets are available, I can look them up myself. Also, we might add or remove datasets, so this would become quickly out of sync.